### PR TITLE
Merge mesh wellcome

### LINKED ISSUE #18: Refactor run_config and run_mesh_config
The functionality in both is almost identical, preprocessing and creating the label binarizer need to be combined in one file that triggers one processs over the other with a flag

### PR DESCRIPTION
**Pull request by nsorros**
Till this point the Wellcome and Mesh tags had some different scripts such as tag, predict and create_label_binarizer. This PR merges them into one project.

Fixes #18

### CODE REVIEW

File: grants_tagger/tag_grants.py [hunk 1]
```diff
@@ -31,15 +25,13 @@ def yield_tagged_grants(grants, scibert_path, tfidf_svm_path, label_binarizer_pa
...
-    grants_tags = predict_tags(
+    tags = predict_tags(
         grants_texts,
         threshold=threshold,
-        scibert_path=scibert_path,
-        tfidf_svm_path=tfidf_svm_path,
+        model_path=model_path,
         label_binarizer_path=label_binarizer_path
     )
     for grant, tags in zip(grants, grants_tags):
```
Comment (lizgzil) [inline grants_tagger/tag_grants.py:37]: something needs to change here since `grant_tags` doesn't exist and you are also rewriting `tags`

**Review by lizgzil — COMMENTED** [review]
mostly looks good and `make tests` pass, a few comments one of which i think is a bug

File: grants_tagger/predict.py [hunk 2]
```diff
@@ -43,62 +44,91 @@ def predict_proba_ensemble_tfidf_svm_bert(X, model_paths):
...
+                      return_probabilities=True, y_batch_size=512):
+    # TODO: generalise tfidf to vectorizer.pkl
+    with open(f"{model_path}/tfidf.pkl", "rb") as f:
+        vectorizer = pickle.loads(f.read())
+
+    Y_pred = []
+    for tag_i in range(0, nb_labels, y_batch_size):
+        with open(f"{model_path}/{tag_i}.pkl", "rb") as f:
+            classifier = pickle.loads(f.read())
+        X_vec = vectorizer.transform(X)
```
[the remark this answers is not in the document]
Comment (nsorros) [reply grants_tagger/predict.py:57]: haha, yet another good point :) yes i can

File: grants_tagger/tag_grants.py (hunk 1 above)
Comment (nsorros) [reply grants_tagger/tag_grants.py:37]: oups, better add a test that can catch this error as well

File: grants_tagger/evaluate_model.py [hunk 3]
```diff
@@ -6,38 +6,13 @@
...
 
 from grants_tagger.utils import load_data
-from grants_tagger.predict_mesh import predict_tfidf_svm, predict_cnn
-from grants_tagger.predict import predict_proba_ensemble_tfidf_svm_bert, load_model
-
-
-def predict(X_test, model_path, nb_labels=None, threshold=0.5, return_probabilities=False):
-    # comma indicates ensemble of more than one models
-    if "," in model_path:
-        model_paths = model_path.split(",")
```
[the remark this answers is not in the document]
Comment (nsorros) [reply grants_tagger/evaluate_model.py:23]: 9a2928a

File: grants_tagger/predict.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nsorros) [reply grants_tagger/predict.py]: 80c8e09

File: grants_tagger/tag_grants.py (hunk 1 above)
Comment (nsorros) [reply grants_tagger/tag_grants.py:46]: f7d9533

**Review by lizgzil — APPROVED** [review]
Looks good `make tests` pass with the new commits :)

**Pull request merged**